sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/request/CreateUserRequest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.request;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request to create a user
+ */
+public class CreateUserRequest extends ActionRequest implements ToXContentObject {
+
+    private String username;
+    private String password;
+
+    public CreateUserRequest(StreamInput in) throws IOException {
+        super(in);
+        username = in.readString();
+        password = in.readString();
+    }
+
+    public CreateUserRequest(String username, String password) {
+        this.username = username;
+        this.password = password;
+    }
+
+    public CreateUserRequest() {}
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (username == null) { // TODO: cehck the condition once
+            validationException = addValidationError("No username specified", validationException);
+        } else if (password == null) {
```

Comment:
What should the response be if both username and password are provided? Should it be `request is missing required parameters` or should it be `request is missing required parameters: username and password`?

Reply:
> What should the response be if both username and password are provided?
It will return null meaning the request is valid.

> Should it be `request is missing required parameters` or should it be `request is missing required parameters: username and password`?
I initially had decided to keep once single message, but then added individual checks to have a clear log in case of validation exception. We can modify this method to add other validations too. (e.g regex validations)